appveyor/install.sh

```diff
@@ -36,6 +36,9 @@ echo "==== get modules"
 $EASYINSTALL pip
 $EASYINSTALL mock
 #$PIP install mock
+$PYTHON -m pip install --upgrade setuptools
+$PYTHON -m pip install --upgrade setuptools_scm
+$PYTHON -m pip install --upgrade git+https://github.com/pypa/pip.git git+https://github.com/pypa/wheel.git
```

Comment:
I would discourage dragging in bleeding edge versions as these could fail the build. Is there any reason to use them?

Reply:
They have very infrequent releases, so if they have broken your stuff, and if it has landed into release, your package will be broken untill the next one.

So you need to know if they have made breaking changes as earlier as possible in order to fix them before they have landed.

There is a yet another reason to install everything reasonable from vcs, not from pip, but it is inapplicable to these packages.

Reply:
I see your point here. This could become a philosophical discussion, but it is pretty much out of our scope to ensure that something that essential like pip is being tested for future versions. If you use that argument, we would also need to test against numpy devel, unreleased Java versions etc. I mean that is the sense of stable APIs we can rely upon, right?

Then a last thing: cloning the whole git history for every CI run we make is overkill. Please get a source archive if you have to use the source for any reason.

Reply:
>If you use that argument, we would also need to test against numpy devel, unreleased Java versions etc. 

If course we would, but building them as part of our build is not reasonable. But if they shipped prebuilt packages, why not.

>cloning the whole git history for every CI run we make is overkill

If I clone manually with `--depth=1` (they refuse to make pip do shallow clones by default https://github.com/pypa/pip/issues/2432 ) it shouldn't be so.

>I mean that is the sense of stable APIs we can rely upon, right?

Not quite. If anything has been broken, it means either the change is explicitly breaking and it is a bug in our package, or the change is not meant to be breaking and it is the bug in their package. Anyway it must be resolved before the release, otherwise users of stable would be very surprized.